ibis/tests/expr/test_format.py

```diff
@@ -197,23 +197,11 @@ def test_memoize_filtered_tables_in_join():
 
 
 def test_argument_repr_shows_name():
-    t = ibis.table([('a', 'int64')], name='t')
-    expr = t.a.nullif(2)
+    t = ibis.table([('fakecolname1', 'int64')], name='fakename2')
+    expr = t.fakecolname1.nullif(2)
     result = repr(expr)
```

Comment:
@cpcloud @saulpw there might be side effects of this PR:
- strange indentation of the table name
- 2 single colons around the column names
- [perhaps] a missing newline from table references

After:

```
ref_0  UnboundTable[table]   t:
    col : int64

Selection[table]
  table:
    Table: ref_0
  selections:
    fakealias1 = Column[int64*] ''col'' from ref_0
    fakealias2 = Column[int64*] ''col'' from ref_0
```

Before:

```
ref_0
UnboundTable[table]
  name: t
  schema:
    col : int64

Selection[table]
  table:
    Table: ref_0
  selections:
    fakealias1 = Column[int64*] 'col' from table
      ref_0
    fakealias2 = Column[int64*] 'col' from table
      ref_0
```

Could we keep explicit output tests at least in `test_format.py`?

Reply:
Hi @kszucs...I feel pretty strongly that we shouldn't test functionality that's not part of the contract.  I made a concession for checking for fragments in `test_format.py` because it does seem useful to make sure the repr has particular meaningful information in it, though I wish it wasn't dependent on the format of that information even in those cases.

I agree that the 2 single quotes around the column name is a negative side effect and I'll fix that (not quite sure how it happened; this should have been almost entirely whitespace).  The removal of newline from table references was intentional, to make the whole repr more compact.  I'm afraid I don't see the "strange indentation" you're talking about though; do you just mean that the table name is on the same line as the table type, now?  That was also intentional, and for the same goal of compactness.

Reply:
I think string indentation may be `col : int64` being indented 4 spaces instead of 2 which is what the rest of the tree uses.

I think instead of `UnboundTable[table]` which has a useless `table` in there, we should just put the table name, like `UnboundTable[name={name}, ref=r{ref_num}]`. This gets rid of the strange `t` and makes the references front and center without adding a newline.

Reply:
Here's an example:

```
r0
UnboundTable[name=t, ref=r0]
  col: int64

Selection[table]
  table:
    ref: r0
  selections:
    fakealias1 = Column[int64*] <- r0.col
    fakealias2 = Column[int64*] <- r0.col
```

Reply:
We can get of the first `r0` too, I just haven't been able to figure out exactly where that's being placed into the final output.

Reply:
Alright, found it:

```
UnboundTable[name=t, ref=r0]
  col: int64

Selection[table]
  table:
    ref: r0
  selections:
    fakealias1 = Column[int64*] <- r0.col
    fakealias2 = Column[int64*] <- r0.col
```

Reply:
Happy to do (or have someone else do) any/all of this. I have several more things I think should be compactified in the repr actually (like ideally, any arg with only one value should have the value on the same line as the arg name).  But I futzed with it and couldn't get it all working like I wanted in a short amount of time and just decided that 30% less vertical whitespace was a solid step, hence this PR.

Reply:
Cool, I'll throw up what I've got so far

Reply:
Also, and this is completely just personal preference, I'm not that much of a fan of `<-` syntax.  I've always had trouble with languages that use it for assignment (it just doesn't look "right" to me), and it's ungreppable (not sure how relevant that is in this case though).